src/network/mprpc/socket.cpp

```diff
@@ -242,6 +242,9 @@ bool socket::set_timeout_sockopt(int sock, int optname, double sec)
 {
   if(sock < 0) { return false; }
 
+  if (sec < 0) {
```

Comment:
In following part, `std::max(0, ...)` do same thing.
So this block is unnecessary? or shall remove `std::max` in following part?